### PR TITLE
Improve the init producer guide

### PR DESCRIPTION
**Pull request by mortent**
Fixes: #877

### CODE REVIEW

File: site/content/en/guides/producer/init/_index.md [hunk 1]
```diff
@@ -74,3 +78,9 @@ Recommended: tag the commit as a release
 git tag nginx/v0.1.0
 git push nginx/v0.1.0
 ```
+
```
Comment (prachirp) [inline site/content/en/guides/producer/init/_index.md:81]: Nit: Update table of contents with new section

File: site/content/en/guides/producer/init/_index.md [hunk 2]
```diff
@@ -8,11 +8,11 @@ description: >
 ---
 
 A kpt package is published as a git subdirectory containing configuration
-files (YAML). Publishes of kpt packages can create or generate YAML files
+files (YAML). Publishers of kpt packages can create or generate YAML files
 however they like using the tool of their choice.
 
 Publishing a package is done by pushing the git directory
-(and optionally tagging it with a version).
+(and optionally [tagging](https://git-scm.com/book/en/v2/Git-Basics-Tagging) it with a version).
```
Comment (prachirp) [inline site/content/en/guides/producer/init/_index.md:15]: Follow the convention of having all links at the bottom of the doc so they can be updated easily.

File: site/content/en/guides/producer/init/_index.md (hunk 1 above)
Comment (mortent) [reply site/content/en/guides/producer/init/_index.md:82]: Fixed

File: site/content/en/guides/producer/init/_index.md (hunk 2 above)
Comment (mortent) [reply site/content/en/guides/producer/init/_index.md]: Done

**Pull request merged**